### DISCUSSION TITLE
Graphql servlet running in Embedded Jetty

### DISCUSSION THREAD

**Discussion by Malte-Neu** [smallrye/smallrye-graphql#1463] [General]
I want to add graphql functionality to my existing embedded jetty project. I found the the graphql servlet, but I can't get it running in my example. The schema generation does not work an the ui schows returns only an error.

Here you can find my test project: https://github.com/Malte-Neu/graphql-servlet-test

I also asked this question a few days ago on stackoverflow (https://stackoverflow.com/questions/72827545), because no one writes an answer I try it here. I hope someone here can show me the way to get it working.

**Comment by phillip-kruger** [comment]
It should work, as we use it in some of our test cases. Maybe have a look there ? What error do you get?

**Comment by phillip-kruger** [comment]
See https://github.com/smallrye/smallrye-graphql/blob/main/server/integration-tests/pom.xml#L68 for example

You can also look at how it's used in the runner project that use wildfly

**Comment by phillip-kruger** [comment]
https://github.com/smallrye/smallrye-graphql/tree/main/server%2Frunner

**Comment by Malte-Neu** [comment]
Thanks for your quick answer.
> It should work, as we use it in some of our test cases. Maybe have a look there ? 

I'll look at the code, I guess it is something wrong with my dependencies. I will compare your code working code with mine.

> What error do you get?

This is the response inside the graphql-ui:
`{
    "errors": [
        "{\n  \"servlet\": \"io.smallrye.graphql.entry.http.ExecutionServlet-2c6a3f77\",\n  \"cause0\": \"java.lang.NullPointerException: Cannot invoke &quot;io.smallrye.graphql.execution.ExecutionService.executeSync(javax.json.JsonObject, io.smallrye.graphql.execution.ExecutionResponseWriter)&quot; because &quot;this.executionService&quot; is null\",\n  \"message\": \"java.lang.NullPointerException: Cannot invoke &quot;io.smallrye.graphql.execution.ExecutionService.executeSync(javax.json.JsonObject, io.smallrye.graphql.execution.ExecutionResponseWriter)&quot; because &quot;this.executionService&quot; is null\",\n  \"url\": \"/graphql\",\n  \"status\": \"500\"\n}"
    ]
}
`

**Comment by phillip-kruger** [comment]
Seem like you are missing CDI

**Comment by Malte-Neu** [comment]
I have tried different things to get CDI working, at least the console logs shows:

> INFO: WELD-ENV-001212: Jetty CdiDecoratingListener support detected, CDI injection will be available in Listeners, Servlets and Filters.

and a small test servlet with and @Inject annotation works. But the inject ExecutionService in the ExecutionServlet is still null. 

My upateded project can be found here https://github.com/Malte-Neu/graphql-servlet-test/tree/weld_v3

**Comment by phillip-kruger** [comment]
Hi @Malte-Neu . I would love to help you more, but I do not have time to spend on this now. Running in Embedded Jetty is not a priority for us. However, we will gladly accept a PR if needed to make it work for you. 

If you look at the pom.xml: https://github.com/smallrye/smallrye-graphql/blob/1.6.x/server/implementation-servlet/pom.xml
Make sure all dependencies listed as provided is in fact part of your jetty server.
(I had a quick look and it seems you have them all)

I also ran your example, and it seems like the StartupListener (https://github.com/smallrye/smallrye-graphql/blob/1.6.x/server/implementation-servlet/src/main/java/io/smallrye/graphql/entry/http/StartupListener.java) is not running on Startup. 

That listener initialize everything, so it should run. You will see things in the log one it run. It bootstrap the graphql-java, using Jandex to find all GraphQL annotations and build the model, and map that to graphql-java and make the schema available.

Happy to debug more with you if you manage to proceed and get the listener to run.

**Comment by phillip-kruger** [comment]
@Malte-Neu - i transferred the issue to a discussion. If you find any issues while trying to get this work, you are welcome to add issues.

**Reply by Malte-Neu** [reply]
Hi @phillip-kruger! I am happy to read that you try to help me. Thanks to point out the problem with the StartupListener, I'll try to find the problem.
